### ISSUE TITLE
Images from Elm files are completely ignored by Parcel

### ISSUE THREAD

**Issue by andreacfromtheapp** [parcel-bundler/parcel#8073]
# 🐛 bug report

I'm using Parcel with Elm and no images from Main.elm are processed into dist. Images from index.html (like favicons) are fine.

## 🎛 Configuration (.babelrc, package.json, cli command)

Standard config following the main Elm guide: [https://parceljs.org/languages/elm/](https://parceljs.org/languages/elm/)

Loading images from Elm looks something like this:

```elm
brand : Html Msg
brand =
    div [ ] [ img [ src "./img/logo.jpg" ] [ ] ]
```

## 🤔 Expected Behavior

I'd expect the image to be "processed" by Parcel and 1) be present in dist, 2) show up on the site.

## 😯 Current Behavior

Images are just ignored. No errors.

## 💁 Possible Solution

Don't have one.

## 🔦 Context

This is a blocking bug, as no images are present at all.

## 💻 Code Sample

WIP repository available here: https://github.com/gacallea/anerandros.info/tree/elm_with_parcel

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.5.0
| Node             | v16.13.2
| Yarn         | 3.2.0
| Operating System | macOS 12.3.1

**Comment by andreacfromtheapp** [comment]
moved repo to github, updated link in issue. here it is just in case: https://github.com/gacallea/anerandros.info/tree/elm_with_parcel

**Comment by mischnic** [comment]
So for regular React, you'd usually have the same problem (so `<img src="x.png">` isn't recognized by Parcel). Instead you need to do this:
https://parceljs.org/recipes/react/#images

The question is how we can do something similar for Elm

**Comment by ChristophP** [comment]
I think it would be possible to add something similar to the way Vite does it https://github.com/hmsk/vite-plugin-elm#static-assets-handling:

One could have a function in elm like this
```elm
-- This function could also be distributed via a package
asset : String -> String
asset src = "PARCEL_ELM_ASSET_PLACEHOLDER:" ++ src
```
It could be used throughout code like this

```elm
  img [ src (asset "./path/to/image.jpeg") ]
```

Then the elm transformer can use `PARCEL_ELM_ASSET_PLACEHOLDER` on the compiled Javascript to replace to find the assets and call `asset.addURLDependency()` for each of them , replacing the url with the generated dependency, which would then be processed by the later stages of bundling.

Is that something that would work?

**Comment by mischnic** [comment]
Doing this text replacement afterwards would theoretically work. It might be better to replace these strings with some existing working expression like `require(...)` or `new URL(...)` though, because the JS transformer that runs after the Elm transformer might be confused if someone added a url dependency beforehand.

So Elm guarantees that `asset` essentially gets inlined into each callsite? Because otherwise you might get `asset("./path/to/image.jpeg")` in the Elm output which wouldn't match with the `PARCEL_ELM_ASSET_PLACEHOLDER` replacement

**Comment by greglearns** [comment]
+1 Checking in to see if there is a solution for this for Elm?
P.S. Thank you for Parcel!

**Comment by hariroshan** [comment]
Hello folks, I wrote a simple asset handler for elm. This will process assets declared in elm.

https://github.com/hariroshan/parcel-transformer-elm-assets

**Comment by ChristophP** [comment]
Hey that's a cool idea of tranforming the output js.👍
Is it even specific to Elm? Seems like it would work for any JS or compile-to-JS languages.

**Comment by hariroshan** [comment]
@ChristophP, yep. It is language agnostic. Just wrote it few hours ago, so I didn’t have time to test it for other languages but it should work.

**Comment by ChristophP** [comment]
Sounds cool, yeah I was just thinking you could even rename it to something more generic like `parcel-transformer-asset-urls`.
